src/platform/forms-system/src/js/validation.js

```diff
@@ -232,10 +232,11 @@ export function errorSchemaIsValid(errorSchema) {
   return _.values(_.omit('__errors', errorSchema)).every(errorSchemaIsValid);
 }
 
-export function isValidForm(form, pageListByChapters) {
-  const pageConfigs = _.flatten(_.values(pageListByChapters));
+export function isValidForm(form, pageList) {
+  const pageListMap = new Map();
+  pageList.forEach(page => pageListMap.set(page.pageKey, page));
```

Comment:
Figured I may as well make getting the `appStateData` a `O(n) + O(kn)` operation (more or less...if I remember right) instead of `O(n^2)` by using a map like this. :man_shrugging: